### PR TITLE
fix: code issues and nits

### PR DESCRIPTION
**Pull request by OnkarRuikar**
The PR fixes
- `substr()` is deprecated, use `substring()`
- `<a>` tag is not closed in one HTML file
- unused local variables
- redundant `;`

### CODE REVIEW

File: static/resources/harness.js [hunk 1]
```diff
@@ -1337,7 +1337,7 @@
       }
 
       if ("serviceWorker" in navigator) {
-        window.__workerCleanup();
+        await window.__workerCleanup();
```
Comment (queengooborg) [inline static/resources/harness.js:1340]: Can't use `async`/`await` in this file; it must stay ES3 compatible.

```suggestion
        window.__workerCleanup();
```

**Review by queengooborg — CHANGES_REQUESTED** [review]

File: views/layout.ejs [hunk 2]
```diff
@@ -43,7 +43,7 @@ See the LICENSE file for copyright details
       <a href="/"><span class="mdi mdi-home"> Home</span></a>
       <a href="/docs/about.md"><span class="mdi mdi-information"> About</span></a>
       <a href="/docs"><span class="mdi mdi-file-document-multiple"> Docs</span></a>
-      <a href="https://opencollective.com/open-web-docs"><span class="mdi mdi-hand-coin"></span> Donate</title>
+      <a href="https://opencollective.com/open-web-docs"><span class="mdi mdi-hand-coin"> Donate</span></a>
```
Comment (queengooborg) [inline views/layout.ejs:46]:
```suggestion
      <a href="https://opencollective.com/open-web-docs"><span class="mdi mdi-hand-coin"></span> Donate</a>
```

Comment (OnkarRuikar) [reply views/layout.ejs]: Your suggestion is not consistent with other items in the navbar. All words are in side `<span>`s.

Comment (queengooborg) [reply views/layout.ejs]: They shouldn't be...

File: scripts/selenium.ts [hunk 3]
```diff
@@ -659,7 +659,7 @@ const run = async (
     if (!ctx.testenv) {
       const filename = path.basename(new URL(downloadUrl).pathname);
       log(task, `Downloading ${filename} ...`);
-      const report = await (await fetch(downloadUrl)).buffer();
+      const report = await (await fetch(downloadUrl)).arrayBuffer();
```
Comment (queengooborg) [inline scripts/selenium.ts:662]: `.arrayBuffer()` is unfortunately not a drop-in replacement for `.buffer()`:

> Argument of type 'ArrayBuffer' is not assignable to parameter of type 'string | ArrayBufferView'.
>  Type 'ArrayBuffer' is missing the following properties from type 'DataView': buffer, byteOffset, getFloat32, getFloat64, and 18 more.

File: views/layout.ejs (hunk 2 above)
Comment (OnkarRuikar) [reply views/layout.ejs:46]: Both render the same though. Ideally if the spans are solely for icons then there shouldn't be any text inside them. Do you want a new PR for this? or piggyback this with one of the next PRs.

Comment (queengooborg) [reply views/layout.ejs:46]: Yeah, the spans are meant solely for the icons -- if you could submit a new PR to move the text out of the spans, I'll merge it right away!

**Pull request merged**